dashboard/test/factories/factories.rb

```diff
@@ -798,6 +798,7 @@
 
   factory :resource do
     url 'fake.url'
+    key 'fake key'
```

Comment:
since key must be unique, I would recommend using a `sequence` here.

I didn't realize this earlier, but key probably will only be able to be unique within the context of a Course Version. I think it's fine to leave the unique index on `key` for now, but we will probably need to change it to be a unique index on (course_version_id, key) once course_version_id is added.